src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
 
     /**
      * Return road accident with matching index
+     *
      * @param index
      * @return
      */
-    public RoadAccident getAccidentByIndex7(String index){
+    public RoadAccident getAccidentByIndex7(String index) {
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            if (current.getAccidentId().equals(index)) {
+                return current;
+            }
+        }
         return null;
     }
 
 
     /**
      * filter list by longtitude and latitude values, including boundaries
+     *
      * @param minLongitude
      * @param maxLongitude
      * @param minLatitude
      * @param maxLatitude
      * @return
      */
-    public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+    public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude) {
+        ArrayList<RoadAccident> result = new ArrayList<>();
+
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            if (minLongitude <= current.getLongitude() && current.getLongitude() <= maxLongitude &&
+                    minLatitude <= current.getLatitude() && current.getLatitude() <= maxLatitude) {
+                result.add(current);
+            }
+        }
+
+        return result;
     }
 
     /**
      * count incidents by road surface conditions
      * ex:
      * wet -> 2
      * dry -> 5
+     *
      * @return
      */
-    public Map<String, Long> getCountByRoadSurfaceCondition7(){
-        return null;
+    public Map<String, Long> getCountByRoadSurfaceCondition7() {
+        HashMap<String, Long> result = new HashMap<>();
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            String roadSurfaceConditions = current.getRoadSurfaceConditions();
+
+            Long newValue = result.get(roadSurfaceConditions) == null ? 1 : Long.valueOf(result.get(roadSurfaceConditions) + 1);
```

Comment:
that was exactly my plan! but so far you are only one who did the same. 
I recommend you to have a look at [this ](https://github.com/Stevenyu163/EPAM-JMP/blob/master/src/main/java/com/epam/processor/DataProcessor.java#L83) solution. Quite interesting use of multimap (I'm not saying it's better then yours - but certainly interesting :)